app/controllers/devise_registration_controller.rb

```diff
@@ -102,7 +122,23 @@ def phone_verify
   def phone_resend
     redirect_to url_for_state and return unless registration_state.state == "phone"
 
-    @phone = registration_state.phone
+    @phone = params[:phone] || registration_state.phone
+
+    return if request.get?
+
+    unless MultiFactorAuth.valid?(params[:phone].presence)
+      @phone_error_message = I18n.t("mfa.errors.phone.invalid")
+      return
+    end
+
+    phone_number = MultiFactorAuth.e164_number(params[:phone] || registration_state.phone)
```

Comment:
We've got both a `@phone` and a `phone_number` in this method?

Reply:
One was formatted internationally, but the other was as entered by the user.  I've updated to include the international number everywhere, since we're reformatting the number when replaying it to the user in the form.